File: 73.Set-Matrix-Zeroes.js
/**
 * @param {number[][]} matrix
 * @return {void} Do not return anything, modify matrix in-place instead.
 */
var setZeroes = function (matrix) {
  const rows = matrix.length;
  const cols = matrix[0].length;
  const zeros = [];

  for (let r = 0; r < rows; r++) {
    for (let c = 0; c < cols; c++) {
      if (matrix[r][c] === 0) {
        zeros.push([r, c]);
      }
    }
  }

  for (const zero of zeros) {
    const [row, col] = zero;

    for (let r = 0; r < rows; r++) {
      matrix[r][col] = 0;
    }

    for (let c = 0; c < cols; c++) {
      matrix[row][c] = 0;
    }
  }
};

File: python/73.Set-Matrix-Zeroes.py
class Solution:
    def setZeroes(self, matrix: List[List[int]]) -> None:
        """
        Do not return anything, modify matrix in-place instead.
        """
        rows = len(matrix)
        cols = len(matrix[0])
        zero_indices = []

        for r in range(rows):
            for c in range(cols):
                if matrix[r][c] == 0:
                    zero_indices.append((r, c))

        for zero_index in zero_indices:
            row, col = zero_index

            for r in range(rows):
                matrix[r][col] = 0
            for c in range(cols):
                matrix[row][c] = 0
